src/cim/components/modals/AddBmcHostYamlForm.tsx

```diff
@@ -0,0 +1,208 @@
+import * as React from 'react';
+import {
+  Alert,
+  AlertActionCloseButton,
+  AlertVariant,
+  Button,
+  ButtonVariant,
+  EmptyState,
+  EmptyStateBody,
+  ModalBoxBody,
+  ModalBoxFooter,
+  Spinner,
+  Text,
+  TextVariants,
+  Title,
+} from '@patternfly/react-core';
+import { UploadActionModalProps } from './types';
+import { ErrorCircleOIcon, UploadIcon } from '@patternfly/react-icons';
+import jsYaml from 'js-yaml';
+import { AddYamlValues } from '../Agent';
+import { Formik, FormikConfig, FormikProps } from 'formik';
+import { saveAs } from 'file-saver';
+import { credentials, host1, host2 } from './constants';
+import './AddBmcHostYamlForm.css';
+import { useTranslation } from '../../../common/hooks/use-translation-wrapper';
+import { getErrorMessage } from '../../../common/utils';
+import { TFunction } from 'i18next';
+
+const emptyValues = (t: TFunction): AddYamlValues => ({
+  fileName: '',
+  fileError: '',
+  textOpenFile: t('ai:Open file'),
+  showOpenFileButton: true,
+  yamlContent: [],
+});
+
+const AddBmcHostYamlForm: React.FC<UploadActionModalProps> = ({ onClose, onCreateBmcByYaml }) => {
+  const [fileName, setFileName] = React.useState<string>();
+  const [fileError, setFileError] = React.useState<string>();
+  const [showOpenFileButton, setShowOpenFileButton] = React.useState(true);
+  // eslint-disable-next-line
+  const [yamlContent, setYamlContent] = React.useState<any>();
+  const [textOpenFile, setTextOpenFile] = React.useState<string>('Open file');
```

Comment:
no need as you have `showOpenFileButton`. Just calculate the `textOpenFile` in code